cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -1146,6 +1146,28 @@ func (csr *ClusterStateRegistry) buildInstanceToErrorCodeMappings(instances []cl
 	return
 }
 
+// GetScaleUpRequests returns map of nodeGroupName -> ScaleUpRequest.
+func (csr *ClusterStateRegistry) GetScaleUpRequests() map[string]*ScaleUpRequest {
+	csr.Lock()
+	defer csr.Unlock()
+	result := make(map[string]*ScaleUpRequest)
+	for nodeGroupId, scaleUpReq := range csr.scaleUpRequests {
+		result[nodeGroupId] = scaleUpReq
```

Comment:
Copying into a new struct is to add a layer of protection in case the caller attempts to change the return value.